### PR TITLE
Persist queries added to workspace

### PR DESCRIPTION
**Pull request by samuelechu**

### CODE REVIEW

**Comment by samuelechu** [comment]
#301

File: src/main/webapp/components/workspace/hooks/use-create-query.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    metacard_tags
+    type
+    sources
+    sorts
+    detail_level
+    id
+  }
+`
+
+export default (onCreate: any) => {
```
Comment (willwill96) [inline src/main/webapp/components/workspace/hooks/use-create-query.tsx]: I like this hook a lot

File: src/main/webapp/components/workspace/workspace.js [hunk 2]
```diff
@@ -94,22 +96,13 @@ export default () => {
...
-        setQueuedSearch('')
-      }
-    },
-    [queries, queuedSearch]
-  )
+  const saveWorkspace = useSaveWorkspace()
+  const createQuery = useCreateQuery(query => {
+    setQueries(merge([query], queries))
+    setCurrentQuery(query.id)
+    saveWorkspace({ queries: merge([query], queries) })
```
Comment (willwill96) [inline src/main/webapp/components/workspace/workspace.js]: :pencil2: could also use es6 syntax here
```suggestion
    saveWorkspace({ queries: [query, ...queries] })
```

File: src/main/webapp/intrigue-api/metacards/metacards.js [hunk 3]
```diff
@@ -460,6 +460,11 @@ const saveMetacard = async (parent, args, context) => {
     )
   }
 
+  if (attributes.queries) {
```
Comment (willwill96) [inline src/main/webapp/intrigue-api/metacards/metacards.js]: :thinking: I think we should consider moving this logic to the react container instead. Normally I would agree that the graphql server should be handling the business logic, but transporting the entire query when you only need the id seems  like a big difference. Do you mind testing the difference on a simulated slow 3g network?

File: src/main/webapp/components/query-manager/add-query.tsx [hunk 4]
```diff
@@ -54,10 +52,9 @@ export default (props: AddQueryProps) => {
         query={query}
         QueryEditor={QueryEditor}
         onSearch={() => {
-          onChange(query)
-          onSearch(query.id!)
+          onCreate(query)
           handleClose()
-          setQuery({ id: new Date().toString() })
+          setQuery({})
```
Comment (willwill96) [inline src/main/webapp/components/query-manager/add-query.tsx]: :question: What is this for?

Comment (samuelechu) [reply src/main/webapp/components/query-manager/add-query.tsx]: We want the add query popover to reset its query state when the query is added

**Pull request merged**